types/payload.go

```diff
@@ -350,7 +349,31 @@ func (p *Payload) extractMetadataFromBytes(data []byte) (int, error) {
 			return len(data) - len(remaining), fmt.Errorf("failed to read value for key %s: %w", string(keyBytes), err)
 		}
 
-		// If we didn't handle this field as metadata, skip it
+		// If not handled as metadata, check if this is a key field
+		// only check for key fields if we haven't found a match yet
+		if !handled && len(samplingKeyFields) > 0 && keysFound < len(samplingKeyFields) {
```

Comment:
You don't need to check `len(samplingKeyFields) > 0`. If `len(samplingKeyFields)` is zero and keysFound is zero we'll still skip this.